iceoryx_dust/include/iceoryx_dust/relocatable_pointer/relocatable_ptr.hpp

```diff
@@ -55,13 +55,17 @@ class relocatable_ptr
     using ptr_t = T*;
 
     /// @brief Construct from raw pointer.
+    // NOLINTJUSTIFICATION implicit conversion from raw pointer is intentional in design of relocatable structures
```

Comment:
I'm a big fan of explicit conversion, but as this is code in `dust` I suppose we can live with it.

Reply:
It makes it otherwise unusable for what it is designed to, at least without creating a wrapper that implicitly converts.

I will show this concept some other time, need to prepare it a little.